### ISSUE TITLE
Indicate in the editor that a type is explicitly `any`

### ISSUE THREAD

**Issue by bpasero** [microsoft/TypeScript#60189]
### 🔍 Search Terms

any marker editor

### ✅ Viability Checklist

- [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
- [x] This wouldn't change the runtime behavior of existing JavaScript code
- [x] This could be implemented without emitting different JS based on the types of the expressions
- [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, new syntax sugar for JS, etc.)
- [x] This isn't a request to add a new utility type: https://github.com/microsoft/TypeScript/wiki/No-New-Utility-Types
- [x] This feature would agree with the rest of our Design Goals: https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals

### ⭐ Suggestion

Types that are explicitly set to `any` are sometimes hard to distinguish from actually typed variables in the editor. You might be adopting types while still having a lot of `any` around and then it would be helpful to call `any` out more explicitly in the editor.

### 📃 Motivating Example

I am now using the extension https://marketplace.visualstudio.com/items?itemName=danvk.any-xray which does this:

![Image](https://github.com/user-attachments/assets/36646902-dc50-4f43-88ac-20c508f21dd0)

### 💻 Use Cases

1. What do you want to use this for?

Seeing explicit `any` types easily so that I can work on adding types.

3. What shortcomings exist with current approaches?

I can disable explicit any by producing errors but that is very aggressive.

5. What workarounds are you using in the meantime?

Hovering over the type.

**Comment by RyanCavanaugh** [comment]
> What workarounds are you using in the meantime?

It seems like you're using the extension, which works?

**Comment by bpasero** [comment]
@RyanCavanaugh yes I do, but there is gaps and I think scalability concerns / hacks in the implementation so I was hoping for an officially TS supported way of getting this information. 

I am also not 100% convinced about the presentation, being forced to see a border around the identifier, something more subtle like this "..." we already see in some cases would be nicer:

![Image](https://github.com/user-attachments/assets/2e7fa21b-49c4-49b7-aa48-b5c14fbdaa09)
